tests/test_overscanCorrection.py

```diff
@@ -347,57 +349,177 @@ def test_parallelOverscanCorrection(self):
 
         This needs to operate on a RawMock() to have overscan data to use.
 
-        The output types may be different when fitType != MEDIAN.
+        This test checks that the outputs match, and that the serial
+        overscan is the trivial value (2.0), and that the parallel
+        overscan is the median of the ramp inserted (4.5)
         """
-        exposure = self.makeExposure(isTransposed=False)
+        exposure = self.makeExposure(addRamp=True, isTransposed=False)
         detector = exposure.getDetector()
         amp = detector.getAmplifiers()[0]
 
         statBefore = computeImageMedianAndStd(exposure.image[amp.getRawDataBBox()])
 
-        config = ipIsr.IsrTask.ConfigClass()
-        config.overscan.doParallelOverscan = True
-        isrTask = ipIsr.IsrTask(config=config)
-        oscanResults = isrTask.overscan.run(exposure, amp)
+        for fitType in ('MEDIAN', 'MEDIAN_PER_ROW'):
+            # This tests these two types to cover scalar and vector
+            # calculations.
+            exposureCopy = exposure.clone()
+            config = ipIsr.IsrTask.ConfigClass()
+            config.overscan.doParallelOverscan = True
+            config.overscan.fitType = fitType
+            isrTask = ipIsr.IsrTask(config=config)
 
-        self.assertIsInstance(oscanResults, pipeBase.Struct)
-        self.assertIsInstance(oscanResults.imageFit, float)
-        self.assertIsInstance(oscanResults.overscanFit, float)
-        self.assertIsInstance(oscanResults.overscanImage, afwImage.ExposureF)
+            oscanResults = isrTask.overscan.run(exposureCopy, amp)
 
-        statAfter = computeImageMedianAndStd(exposure.image[amp.getRawDataBBox()])
-        self.assertLess(statAfter[0], statBefore[0])
+            self.assertIsInstance(oscanResults, pipeBase.Struct)
+            if fitType == 'MEDIAN':
+                self.assertIsInstance(oscanResults.imageFit, float)
+                self.assertIsInstance(oscanResults.overscanFit, float)
+            else:
+                self.assertIsInstance(oscanResults.imageFit, np.ndarray)
+                self.assertIsInstance(oscanResults.overscanFit, np.ndarray)
+            self.assertIsInstance(oscanResults.overscanImage, afwImage.ExposureF)
+
+            statAfter = computeImageMedianAndStd(exposureCopy.image[amp.getRawDataBBox()])
+            self.assertLess(statAfter[0], statBefore[0])
 
-    def test_badParallelOverscanCorrection(self):
+            # Test the output value for the serial and parallel overscans
+            self.assertEqual(oscanResults.overscanMean[0], 2.0)
+            self.assertEqual(oscanResults.overscanMean[1], 4.5)
```

Comment:
These tests are selling the correction short.  I looked at the ramp image and the `MEDIAN` and `MEDIAN_PER_ROW` corrections, and the `MEDIAN_PER_ROW` removes the ramp (and drives `statAfter[1]` to 0.0).  (The `MEDIAN` does not fix this).  So I think that there should be an additional test in the `MEDIAN_PER_ROW` case that checks that the stddev from `statAfter` either goes to zero or at least is reduced by the parallel overscan correction.